Libraries/pbxbuild/Sources/Phase/Context.cpp

```diff
@@ -323,24 +324,28 @@ resolveBuildFiles(
 
             if (buildRule != nullptr) {
                 if (pbxspec::PBX::Tool::shared_ptr const &tool = buildRule->tool()) {
-                    // Some tools additionally limit their file types beyond what their build rule allows.
-                    // For example, the default compiler limits itself to just source files, despite its
-                    // default build rule specifying that it accepts all C-family inputs, including headers.
+                    /*
+                     * Some tools additionally limit their file types beyond what their build rule allows.
+                     * For example, the default compiler limits itself to just source files, despite its
+                     * default build rule specifying that it accepts all C-family inputs, including headers.
```

Comment:
The issue is that it takes `sourcecode.cpp`, and `sourcecode.cpp.header` inherits from `sourcecode.cpp`. I tried not checking the inheritance hierarchy, but that failed for `text.plist.xml` inherits from `text.plist` and needs to be an input to the plist copy tool...